tests/framework/crossplane.go

```diff
@@ -64,6 +66,8 @@ func ValidateNginxFieldExists(conf *Payload, expFieldCfg ExpectedNginxField) err
 					}
 				}
 			}
+
+			return validateUpstreamDirectives(expFieldCfg, directive)
```

Comment:
I think we only want to validate the upstream directives if `expFieldCfg.Upstreams` is specified.